### ISSUE TITLE
Parallax Component does not work with more than 3 images

### ISSUE THREAD

**Issue by tiofabby** [flame-engine/flame#1154]
Hi,
I am running flame: ^1.0.0-releasecandidate.17 and implementing parallax component with 3 images with following code, it works fine. Images are running faster from background to front.

If I uncomment commented lines and add one or 2 images, velocity gets weird and background goes faster than front images!

Code here:
```
class MyParallaxComponent extends ParallaxComponent {
  @override
  Future<void> onLoad() async {
    parallax = await gameRef.loadParallax([
      ParallaxImageData('parallax-background/spring/bgr_spring1.png'),
      ParallaxImageData('parallax-background/spring/bgr_spring2.png'),
      ParallaxImageData('parallax-background/spring/bgr_spring3.png'),
      // ParallaxImageData('parallax-background/spring/bgr_spring4.png'),
      // ParallaxImageData('parallax-background/spring/bgr_spring5.png')
    ],
        baseVelocity: Vector2(10, 0),
        velocityMultiplierDelta: Vector2(1.3, 1.0));
  }
}

class MyGame extends FlameGame {
  @override
  Future<void> onLoad() async {
    await super.onLoad();
    add(MyParallaxComponent());
  }
}
```

**Comment by spydon** [comment]
Hi,
the examples are also running with 5 images and I can't see any problems in those ones:
https://examples.flame-engine.org/#/Parallax_Basic

Are your images of the same size?

**Comment by tiofabby** [comment]
Hi @spydon
I tried with the images from the example you provided: (https://github.com/flame-engine/flame/tree/main/examples/assets/images/parallax) and I saw that the 2 first background images are half width compared to the 3 other front images.
So I tried to do the same for my own images: I cut half width my 2 first background images and... it worked!
So I guess this could be a temporary workaround to fix such issue... as all images should be same size and velocity should work correctly.
Thank you
Fabrice

**Comment by spydon** [comment]
You shouldn't need to cut the first images, I just tried with 5 identical images, and it works as intended.
Could you share a minimal reproducible example in a repository that I can run?

**Comment by tiofabby** [comment]
Hi @spydon , First, Thank you for spending time on such issue. I created then a very small repo running parallax for the 5 images I have on my side:

https://github.com/tiofabby/debugflame.git

When I run it, background 3 and 4 are going slower than 1, 2 and 5 which makes parallax effect kind of weird.
Thank you

**Comment by spydon** [comment]
Alright, I think I know what is happening.
The images are loading asynchronously and they are getting their speed when they are done loading, instead of the order they are in.
A workaround for this can be to preload the images:
```dart
    await gameRef.images.loadAll([
      'bgr_spring1.png',
      'bgr_spring2.png',
      'bgr_spring3.png',
      'bgr_spring4.png',
      'bgr_spring5.png'
    ]);
```
So it is indeed a bug that we need to fix. :)

**Comment by spydon** [comment]
I fixed it in #1155, if you want to try out the fix you can depend on that branch like this:
```yaml
  flame:
    git:
      url: https://github.com/flame-engine/flame.git
      ref: spydon.fix-parallax-loading
      path: packages/flame
```
(instead of the current Flame version that you have in your `pubspec.yaml`)

**Comment by tiofabby** [comment]
Hi @spydon , your fix works fine!! 👍 Thank you!!
